test/test_timesteparrays.jl

```diff
@@ -97,12 +97,6 @@ x[t3] = temp_dim_val[1]
 @test x[t3] == temp_dim_val[1]
 reset_time_val(x, time_dim_val)
 
-# Deprecated int indexing should still run
-@test x[3] == time_dim_val[3]
-x[TimestepIndex(3)] = temp_dim_val[3]
-@test x[TimestepIndex(3)] == temp_dim_val[3]
-reset_time_val(x, time_dim_val)
-
```

Comment:
should we leave part of this test to test that this now errors? (might help with our coverage of the functions we have deprecated, but the code is still there?)

Reply:
I guess this is a broader question about coverage of deprecated functions. @davidanthoff is there a way for us to list certain functions that the coverage tests don't care about? because this PR is definitely going to decrease our testing coverage unfortunately

Reply:
Yea, it is going to decrease coverage.  The other option I guess is to add tests for errors and then remove them when we remove the specific error messages in a later release I just wasn't sure if that was overkill and wanted to keep testing clean ... but coverage then goes down.

Reply:
`@test_throws` seems the right approach for this kind of situation?

Reply:
@davidanthoff I added back in the tests for old behavior that were already there, now with `@test_throws`. Now _overall_ test coverage for Mimi has increased, but the _diff_ coverage is not 100% because there were some deprecated Int-indexing functions that were not previously being tested, and that this PR changed from warnings to errors. Do you want me to add individual tests for each of these, or should we ignore it?

Reply:
Lets ignore that